### PR TITLE
 Fix #1346: Add feature to invite users to participant team using URL

### LINKED ISSUE #1346: Add feature to invite users to participant team using URL
## Deliverables

Add a feature to invite new team member to a participant team using a Unique URL. Also, don't forget to write unit tests. :) 

## Steps

1. **user1** enters the email of his teammate (let's say **user2**)
2. Send an email to **user2** (if he/she is registered on EvalAI) with a unique URL
3. **user2** clicks on the URL to accept the invitation to the team.
4. *user1* gets an email saying that "**user2** has accepted your invite to team <blah>" (we can iterate over the email message later) 

### Some important points to note

- [ ] Checks performed before inviting to the team should be similar to current invitation logic

### PR DESCRIPTION
**Pull request by Sanji515**
**Deliverables**

- [x] Added a feature to invite new team member to a participant team using a Unique URL.
- [x] Written unit tests.

Here's a screencast of it->

recording 5

### CODE REVIEW

**Comment by Sanji515** [comment]
@RishabhJain2018 @deshraj Please review.

**Comment by kurianbenoy** [comment]
@Sanji515 the test coverage has decreased. It would be good if you can write tests for your new feature you have added in views.py

**Comment by Sanji515** [comment]
@kurianbenoy I've added some more test for sending mail, still it's failing... why it's failing now.

**Comment by Sanji515** [comment]
@kurianbenoy I've added some more tests for sending the mail, why it's failing now?

**Comment by kurianbenoy** [comment]
https://coveralls.io/builds/21824812/source?filename=apps/participants/serializers.py The test coverage is being shown decreased in `serializers.py` file. It seems a bit strange to me. I will check out in more depth and reply back soon @Sanji515

**Comment by kurianbenoy** [comment]
https://coveralls.io/builds/21824812/source?filename=apps/participants/serializers.py The test coverage is being shown decreased in `serializers.py` file. It seems a bit strange to me. I will check out in more depth and reply back soon @Sanji515

File: apps/participants/templates/participant_team_email_invite.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+You've been invited to join the team {{ team_name }}. Click the bottom link to accept the invitation.
+{{ full_url }}
```
Comment (Sanji515) [inline apps/participants/templates/participant_team_email_invite.html]: @RishabhJain2018 I've already added a new line, this is my latest commit.

**Comment by Sanji515** [comment]
@RishabhJain2018 @galipremsagar Please review.

**Comment by galipremsagar** [comment]
> @galipremsagar, please review

Will look at this today @Sanji515

**Comment by RishabhJain2018** [comment]
@galipremsagar @rishabhbhardwaj You guys are taking care of this PR, right?

File: apps/participants/views.py [hunk 2]
```diff
@@ -197,18 +202,83 @@ def invite_participant_to_team(request, pk):
...
-        }
+    '''
+    Sending unique email url to accept the invitation
+    '''
+    team_name = participant_team.team_name
+    encoded_team_id = urlsafe_base64_encode(force_bytes(pk)).decode()
+    encoded_email = urlsafe_base64_encode(force_bytes(email)).decode()
+    combined_url = "{}/{}".format(encoded_team_id, encoded_email)
+    web_url = request.data['url'].split("team")[0]
+    full_url = "{}invitation/{}".format(web_url, combined_url)
```
Comment (galipremsagar) [inline apps/participants/views.py]: See how hostname is being extracted from previously written functions. 
https://github.com/Cloud-CV/EvalAI/blob/a10d5275e9d0cdba0e4dcde7cfb5ae8def5ac82d/apps/challenges/views.py#L1735

**Comment by galipremsagar** [comment]
Also, see this workflow:
1. user1 invites user2 to join a team. 
2. user2 gets a url.
3. user2 opens the url in a browser where he isn't signed in, he is redirected to login page.
4. user2 logs-in, and he **should** see a success message that he has been added to team. **But currently it is not the case**. This is the error message I'm seeing is:
Screenshot 2019-05-11 at 12 16 21 AM
5. After I re-open the same url when user2 is logged in only I'm being added successfully:
Screenshot 2019-05-11 at 12 16 36 AM

**Please fix 4.**

File: frontend/src/js/controllers/teamsCtrl.js [hunk 3]
```diff
@@ -361,6 +362,36 @@
...
+                        $rootScope.notify("error",
+                            'Team invitation is not yet accepted, please go to the link again.');
+                    }
+                    else{
+                        $rootScope.notify("error", error);
+                    }
+                }
+            };
+            utilities.sendRequest(parameters);
+
```
Comment (galipremsagar) [inline frontend/src/js/controllers/teamsCtrl.js]: Please remove extra new lines

**Pull request closed without merging**